src/storage/src/hummock/store/version.rs

```diff
@@ -67,29 +68,23 @@ impl StagingVersion {
     pub fn prune_overlap<'a>(
         &'a self,
         epoch: HummockEpoch,
-        compaction_group_id: CompactionGroupId,
+        table_id: TableId,
         key_range: &'a (Bound<Vec<u8>>, Bound<Vec<u8>>),
     ) -> (
         impl Iterator<Item = &ImmutableMemtable> + 'a,
         impl Iterator<Item = &SstableInfo> + 'a,
     ) {
         let overlapped_imms = self.imm.iter().filter(move |imm| {
-            (compaction_group_id
-                == StaticCompactionGroupId::NewCompactionGroup as CompactionGroupId
-                || compaction_group_id == imm.compaction_group_id())
-                && imm.epoch() <= epoch
+            imm.epoch() <= epoch
```

Comment:
We may also filter by whether table_id matches.

Reply:
> We may also filter by whether table_id matches.

I think that range overlap for imm here indicates that table_id match